Climate revamp

Dev2150 opened the pull request:
[Discord thread link](https://discord.com/channels/278953791586304005/1445061119155896450)

Dev2150 committed:
feat(road): Estates & government pay for road maintenance

Dev2150 committed:
feat(road): Add maintenance cost based on terrain types

Dev2150 committed:
feat(road): Add maintenance cost map mode

Dev2150 committed:
feat(road): Display road maintenance & show actual expense & balance in the budget

Dev2150 committed:
feat(climate): Add localization for new climate system

Dev2150 committed:
feat(climate): Add named colors for new climate system

Dev2150 committed:
feat(climate): Add new climates

Dev2150 committed:
feat(climate): Add `free_capacity_attracts_pops`

Dev2150 committed:
feat(climate): Add location_templates.txt

Dev2150 committed:
Merge remote-tracking branch 'upstream/develop' into climate_revamp

# Conflicts:
#	Documentation/Change log.md
#	in_game/common/on_action/MnT_pulse.txt
#	in_game/common/scripted_effects/SYS-scripted_effect.txt
#	in_game/gfx/map/map_modes/mnt_map_modes.txt
#	in_game/gui/economy_lateralview.gui
#	main_menu/common/game_concepts/MnT_game_concepts.txt
#	main_menu/localization/english/MnT_game_concepts_l_english.yml
#	main_menu/localization/english/MnT_interfaces_l_english.yml

Dev2150 committed:
fix(economy): Add BOM to files

aldaron-MEIOU committed:
Renamed all keys to more easily identified names

aldaron-MEIOU committed:
Added same RGB colors are Koppen has

aldaron-MEIOU committed:
Added localizations of new climate names and their descriptions

aldaron-MEIOU committed:
Modified a few climates here and there for fine tuning

aldaron-MEIOU committed:
Saturated colors of Cfa, CSb and Bsk for colorblind people. Also added Andean Tundra Climate

Malorn-Deslor commented:
Hey guys, we're at the point where we need the climates to assign food buildings. How's it looking?

aldaron-MEIOU committed:
Merge branch 'develop' into climate_revamp

# Conflicts:
#	Documentation/Change log.md
#	in_game/common/on_action/MnT_pulse.txt
#	in_game/gfx/map/map_modes/mnt_map_modes.txt
#	main_menu/localization/english/MnT_interfaces_l_english.yml

cheetawilliam committed:
new climates and rebalanced topography and vegitation

https://docs.google.com/spreadsheets/d/1ZY1k7X1LymSPdtjcGXBNXomkm0Gz-th-1-Q6PoS4nbY/edit?usp=sharing
See this doc for tables with all the values and reasoning for min /max values

cheetawilliam committed:
remove food effect from winter

cheetawilliam committed:
use REPLACE

cheetawilliam committed:
being stupid

cheetawilliam committed:
Merge branch 'develop' into climate_revamp

cheetawilliam committed:
fix merge conflicts

cheetawilliam committed:
adjust hot steppe

make hot steppe not worse than hot desert for pop cap, both at 0 now.

MichaHelder committed:
fix(error): fix modifier type file name being capitalized

doesn't work if that is the case
edit: just rebuild the whole file
